### PR TITLE
FE-317 User can make in-app unsuspension requests

### PR DESCRIPTION
**Pull request by jonambas**
Jira ticket: https://jira.int.messagesystems.com/browse/FE-317

---

New component: `<SuspensionAlerts />`  rendered under `<App />`

- Moved all code around these alerts from AuthGate and globalAlert actions into this component.
- It felt weird to render jsx inside an action
- All logic is now in one place

screen shot 2018-05-08 at 2 54 46 pm

### CODE REVIEW

File: src/components/auth/SuspensionAlerts.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    return this.props.isSuspendedForBilling
+      ? (
+        <Fragment>
+          <div>Your account is currently suspended due to a billing problem.</div>
+          <div>To make a payment and reactivate your account, <UnstyledLink component={Link} to='/account/billing'>visit the billing page</UnstyledLink>.</div>
+        </Fragment>
+      ) : (
+        <Fragment>
+          <div>Your account is currently suspended.</div>
+          <div>For any questions or to request reactivation, please <UnstyledLink onClick={this.openTicket}>submit a ticket</UnstyledLink>.</div>
```
Comment (jonambas) [inline src/components/auth/SuspensionAlerts.js]: This currently poses a small problem. Free accounts will still see this alert, but clicking this link won't render the support UI.

File: src/components/auth/SuspensionAlerts.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    }
+  }
+
+  render () {
+    return null;
+  }
+}
+
+const mapStateToProps = (state) => ({
+  status: state.account.status,
```
Comment (bkemper) [inline src/components/auth/SuspensionAlerts.js]: Should we used the [`hasStatus()`](https://github.com/SparkPost/2web2ui/blob/master/src/helpers/conditions/account.js#L10) helper?

File: src/components/auth/SuspensionAlerts.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    this.props.openSupportPanel({ view: 'ticket' });
+    this.props.hydrateTicketForm({ issueId: 'account_suspension' });
+  }
+
+  getMessage () {
+    return this.props.isSuspendedForBilling
+      ? (
+        <Fragment>
+          <div>Your account is currently suspended due to a billing problem.</div>
+          <div>To make a payment and reactivate your account, <UnstyledLink component={Link} to='/account/billing'>visit the billing page</UnstyledLink>.</div>
```
Comment (bkemper) [inline src/components/auth/SuspensionAlerts.js]: `<PageLink />` could be used here.

File: src/components/auth/tests/SuspensionAlerts.test.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+      isSuspendedForBilling: false,
+      openSupportPanel: jest.fn(),
+      hydrateTicketForm: jest.fn()
+    };
+
+    wrapper = shallow(<SuspensionAlerts {...props} />);
+  });
+
+  afterEach(() => {
+    jest.resetAllMocks();
```
Comment (bkemper) [inline src/components/auth/tests/SuspensionAlerts.test.js]: This can be removed.

**Pull request merged**